src/linear_algebra/char_poly/coeff.lean

```diff
@@ -160,3 +162,35 @@ begin
   rw [coeff_zero_eq_eval_zero, char_poly, eval_det, mat_poly_equiv_char_matrix, ← det_smul],
   simp
 end
+
+variables {p : ℕ} [fact p.prime]
+
+lemma zmod.char_poly_pow_card_of_nonempty [nonempty n] (M : matrix n n (zmod p)) :
```

Comment:
I have a hard time believing that this lemma could be false when `n` is empty, if only because then surely `M ^ p = M`.